src/Stratis.Bitcoin.Features.Consensus/Behaviors/ProvenHeadersConsensusManagerBehavior.cs

```diff
@@ -123,12 +122,16 @@ protected override Payload ConstructHeadersPayload(GetHeadersPayload getHeadersP
                 return null;
             }
 
-            var headers = new ProvenHeadersPayload();
-            foreach (ChainedHeader header in this.chain.EnumerateToTip(fork).Skip(1))
+            var provenHeadersPayload = new ProvenHeadersPayload();
+
+            ChainedHeader header = this.FindStartingHeader(fork, getHeadersPayload.HashStop);
```

Comment:
Maybe rename to FindLastHeader ? 
Because it's not starting header, its the last header to include in the response payload.